mkpj panics when more than one job has the same name

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!-->


**What happened**:
```
mkpj --job integ-framework-k8s --config-path config.yaml
WARN[0000] empty -github-token-path, will use anonymous github client
panic: runtime error: index out of range

goroutine 1 [running]:
main.(*options).defaultPR(0xc0005411e8, 0xc000540f58, 0xc000055b90, 0x0)
        /usr/local/google/home/howardjohn/go/src/k8s.io/test-infra/prow/cmd/mkpj/main.go:70 +0x2b8
main.main()
        /usr/local/google/home/howardjohn/go/src/k8s.io/test-infra/prow/cmd/mkpj/main.go:256 +0x145f
```

**What you expected to happen**:
`mkpj` doesn't panic or fails graceful and indicates what I need to do. Is github token path required? Am I missing some other flag?

I installed with `go get -u k8s.io/test-infra/prow/cmd/mkpj`
The code crashing is the first line of this:

```go

func (o *options) defaultPR(pjs *prowapi.ProwJobSpec) error {
	if pjs.Refs.Pulls[0].Number == 0 { // CRASH HERE
		fmt.Fprint(os.Stderr, "PR Number: ")
		var pullNumber int
		fmt.Scanln(&pullNumber)
		pjs.Refs.Pulls[0].Number = pullNumber
		o.pullNumber = pullNumber
	}
	if pjs.Refs.Pulls[0].Author == "" {
		pr, err := o.getPullRequest()
		if err != nil {
			return err
		}
		pjs.Refs.Pulls[0].Author = pr.User.Login
	}
	if pjs.Refs.Pulls[0].SHA == "" {
		pr, err := o.getPullRequest()
		if err != nil {
			return err
		}
		pjs.Refs.Pulls[0].SHA = pr.Head.SHA
	}
	return nil
}
```

--- a/prow/cmd/mkpj/main.go
+++ b/prow/cmd/mkpj/main.go
@@ -92,9 +92,6 @@ func (o *options) defaultPR(pjs *prowapi.ProwJobSpec) error {
 }
 
 func (o *options) defaultBaseRef(pjs *prowapi.ProwJobSpec) error {
-	if pjs.Refs == nil {
-		return nil
-	}
 	if pjs.Refs.BaseRef == "" {
 		if o.pullNumber != 0 {
 			pr, err := o.getPullRequest()
@@ -191,8 +188,6 @@ func main() {
 	var labels map[string]string
 	var annotations map[string]string
 	var found bool
-	var needsBaseRef bool
-	var needsPR bool
 	for fullRepoName, ps := range conf.Presubmits {
 		org, repo, err := splitRepoName(fullRepoName)
 		if err != nil {
@@ -215,8 +210,6 @@ func main() {
 				labels = p.Labels
 				annotations = p.Annotations
 				found = true
-				needsBaseRef = true
-				needsPR = true
 			}
 		}
 	}
@@ -237,7 +230,6 @@ func main() {
 				labels = p.Labels
 				annotations = p.Annotations
 				found = true
-				needsBaseRef = true
 			}
 		}
 	}
@@ -255,13 +247,11 @@ func main() {
 	if pjs.Refs != nil {
 		o.org = pjs.Refs.Org
 		o.repo = pjs.Refs.Repo
-	}
-	if needsPR {
-		if err := o.defaultPR(&pjs); err != nil {
-			logrus.WithError(err).Fatal("Failed to default PR")
+		if len(pjs.Refs.Pulls) != 0 {
+			if err := o.defaultPR(&pjs); err != nil {
+				logrus.WithError(err).Fatal("Failed to default PR")
+			}
 		}
-	}
-	if needsBaseRef {
 		if err := o.defaultBaseRef(&pjs); err != nil {
 			logrus.WithError(err).Fatal("Failed to default base ref")
 		}

--- a/prow/cmd/mkpj/main.go
+++ b/prow/cmd/mkpj/main.go
@@ -54,6 +54,54 @@ type options struct {
 	pullRequest  *github.PullRequest
 }
 
+func (o *options) genJobSpec(conf *config.Config, name string) (config.JobBase, prowapi.ProwJobSpec) {
+	for fullRepoName, ps := range conf.Presubmits {
+		org, repo, err := splitRepoName(fullRepoName)
+		if err != nil {
+			logrus.WithError(err).Warnf("Invalid repo name %s.", fullRepoName)
+			continue
+		}
+		for _, p := range ps {
+			if p.Name == o.jobName {
+				return p.JobBase, pjutil.PresubmitSpec(p, prowapi.Refs{
+					Org:     org,
+					Repo:    repo,
+					BaseRef: o.baseRef,
+					BaseSHA: o.baseSha,
+					Pulls: []prowapi.Pull{{
+						Author: o.pullAuthor,
+						Number: o.pullNumber,
+						SHA:    o.pullSha,
+					}},
+				})
+			}
+		}
+	}
+	for fullRepoName, ps := range conf.Postsubmits {
+		org, repo, err := splitRepoName(fullRepoName)
+		if err != nil {
+			logrus.WithError(err).Warnf("Invalid repo name %s.", fullRepoName)
+			continue
+		}
+		for _, p := range ps {
+			if p.Name == o.jobName {
+				return p.JobBase, pjutil.PostsubmitSpec(p, prowapi.Refs{
+					Org:     org,
+					Repo:    repo,
+					BaseRef: o.baseRef,
+					BaseSHA: o.baseSha,
+				})
+			}
+		}
+	}
+	for _, p := range conf.Periodics {
+		if p.Name == o.jobName {
+			return p.JobBase, pjutil.PeriodicSpec(p)
+		}
+	}
+	return config.JobBase{}, prowapi.ProwJobSpec{}
+}
+
 func (o *options) getPullRequest() (*github.PullRequest, error) {
 	if o.pullRequest != nil {
 		return o.pullRequest, nil
@@ -183,56 +231,7 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatal("Failed to get GitHub client")
 	}
-
-	var job config.JobBase
-	var pjs prowapi.ProwJobSpec
-	for fullRepoName, ps := range conf.Presubmits {
-		org, repo, err := splitRepoName(fullRepoName)
-		if err != nil {
-			logrus.WithError(err).Warnf("Invalid repo name %s.", fullRepoName)
-			continue
-		}
-		for _, p := range ps {
-			if p.Name == o.jobName {
-				job = p.JobBase
-				pjs = pjutil.PresubmitSpec(p, prowapi.Refs{
-					Org:     org,
-					Repo:    repo,
-					BaseRef: o.baseRef,
-					BaseSHA: o.baseSha,
-					Pulls: []prowapi.Pull{{
-						Author: o.pullAuthor,
-						Number: o.pullNumber,
-						SHA:    o.pullSha,
-					}},
-				})
-			}
-		}
-	}
-	for fullRepoName, ps := range conf.Postsubmits {
-		org, repo, err := splitRepoName(fullRepoName)
-		if err != nil {
-			logrus.WithError(err).Warnf("Invalid repo name %s.", fullRepoName)
-			continue
-		}
-		for _, p := range ps {
-			if p.Name == o.jobName {
-				job = p.JobBase
-				pjs = pjutil.PostsubmitSpec(p, prowapi.Refs{
-					Org:     org,
-					Repo:    repo,
-					BaseRef: o.baseRef,
-					BaseSHA: o.baseSha,
-				})
-			}
-		}
-	}
-	for _, p := range conf.Periodics {
-		if p.Name == o.jobName {
-			job = p.JobBase
-			pjs = pjutil.PeriodicSpec(p)
-		}
-	}
+	job, pjs := o.genJobSpec(conf, o.jobName)
 	if job.Name == "" {
 		logrus.Fatalf("Job %s not found.", o.jobName)
 	}

--- a/prow/cmd/mkpj/main.go
+++ b/prow/cmd/mkpj/main.go
@@ -184,10 +184,8 @@ func main() {
 		logrus.WithError(err).Fatal("Failed to get GitHub client")
 	}
 
+	var job config.JobBase
 	var pjs prowapi.ProwJobSpec
-	var labels map[string]string
-	var annotations map[string]string
-	var found bool
 	for fullRepoName, ps := range conf.Presubmits {
 		org, repo, err := splitRepoName(fullRepoName)
 		if err != nil {
@@ -196,6 +194,7 @@ func main() {
 		}
 		for _, p := range ps {
 			if p.Name == o.jobName {
+				job = p.JobBase
 				pjs = pjutil.PresubmitSpec(p, prowapi.Refs{
 					Org:     org,
 					Repo:    repo,
@@ -207,9 +206,6 @@ func main() {
 						SHA:    o.pullSha,
 					}},
 				})
-				labels = p.Labels
-				annotations = p.Annotations
-				found = true
 			}
 		}
 	}
@@ -221,27 +217,23 @@ func main() {
 		}
 		for _, p := range ps {
 			if p.Name == o.jobName {
+				job = p.JobBase
 				pjs = pjutil.PostsubmitSpec(p, prowapi.Refs{
 					Org:     org,
 					Repo:    repo,
 					BaseRef: o.baseRef,
 					BaseSHA: o.baseSha,
 				})
-				labels = p.Labels
-				annotations = p.Annotations
-				found = true
 			}
 		}
 	}
 	for _, p := range conf.Periodics {
 		if p.Name == o.jobName {
+			job = p.JobBase
 			pjs = pjutil.PeriodicSpec(p)
-			labels = p.Labels
-			annotations = p.Annotations
-			found = true
 		}
 	}
-	if !found {
+	if job.Name == "" {
 		logrus.Fatalf("Job %s not found.", o.jobName)
 	}
 	if pjs.Refs != nil {
@@ -256,7 +248,7 @@ func main() {
 			logrus.WithError(err).Fatal("Failed to default base ref")
 		}
 	}
-	pj := pjutil.NewProwJob(pjs, labels, annotations)
+	pj := pjutil.NewProwJob(pjs, job.Labels, job.Annotations)
 	b, err := yaml.Marshal(&pj)
 	if err != nil {
 		logrus.WithError(err).Fatal("Error marshalling YAML.")

--- a/prow/cmd/mkpj/main.go
+++ b/prow/cmd/mkpj/main.go
@@ -217,8 +217,6 @@ func main() {
 				found = true
 				needsBaseRef = true
 				needsPR = true
-				o.org = org
-				o.repo = repo
 			}
 		}
 	}
@@ -240,8 +238,6 @@ func main() {
 				annotations = p.Annotations
 				found = true
 				needsBaseRef = true
-				o.org = org
-				o.repo = repo
 			}
 		}
 	}
@@ -255,6 +251,10 @@ func main() {
 	}
 	if !found {
 		logrus.Fatalf("Job %s not found.", o.jobName)
+	}
+	if pjs.Refs != nil {
+		o.org = pjs.Refs.Org
+		o.repo = pjs.Refs.Repo
 	}
 	if needsPR {
 		if err := o.defaultPR(&pjs); err != nil {

